process/block/shardblock.go

```diff
@@ -262,6 +262,13 @@ func (sp *shardProcessor) ProcessBlock(
 		return err
 	}
 
+	if header.IsStartOfEpochBlock() {
+		err = sp.checkEpochCorrectnessCrossChain(chainHandler)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Do you forgot to add this line after checking the err? sp.epochStartTrigger.SetProcessed(header)